crates/resolver/src/config.rs

```diff
@@ -913,6 +842,13 @@ pub struct ResolverOpts {
     /// prevent those prompts from being displayed. If os_port_selection is true, avoid_local_udp_ports
     /// will be ignored.
     pub os_port_selection: bool,
+    /// Optional configuration for the TLS client.
+    ///
+    /// The correct ALPN for the corresponding protocol is automatically
+    /// inserted if none was specified.
+    #[cfg(feature = "dns-over-rustls")]
+    #[cfg_attr(feature = "serde", serde(skip))]
+    pub tls_config: Option<Arc<rustls::ClientConfig>>,
```

Comment:
by moving this here, doesn't it imply that we can only have a single TLS config per Resolver instance? With it as part of the NameServerConfig it allows for a TLS config per Nameserver. Is this a desired from that perspective?

Reply:
Yes, it does. Do you recall use cases where this would be an issue? It would be a lot more efficient to only have one TLS config per `Resolver` instance and I think that covers the vast majority of usage, but it's possible there are edge cases where it wouldn't work well. (Either way, I think `NameServerConfigGroup` was an odd place for it so I think we want to do something different.

Propose we try this and see if someone comes out of the woodwork to demand more flexible configuration? Should be easy to treat this as a fallback and allow more specific configuration in other places.

Reply:
I think the reason I originally did it on each name server config was with the idea that we could present different client certs if people wanted to. Does the TLS config allow for different client chains to be registered for different servers?

this might be a super edge condition and not important to support, but it was the biggest reason I could think of.

Reply:
It is theoretically possible to resolve different client auth certificate chains IIRC, but if this is only a theoretical use case (and arguably pretty niche) I propose to ditch it for now and see if anyone complains. Can always add back API to make it more fine-grained.

Reply:
Ok. Thanks for the discussion both of you. I’m convinced.